Gems/Atom/Tools/MaterialCanvas/Code/Source/Document/MaterialCanvasDocument.cpp

```diff
@@ -793,67 +971,108 @@ namespace MaterialCanvas
+    AZStd::string MaterialCanvasDocument::GetSymbolNameFromNode(GraphModel::ConstNodePtr node) const
     {
-        return AtomToolsFramework::GetSymbolNameFromText(AZStd::string::format("node%u_%s", inputNode->GetId(), inputNode->GetTitle()));
+        return AtomToolsFramework::GetSymbolNameFromText(AZStd::string::format("node%u_%s", node->GetId(), node->GetTitle()));
+    }
+
+    AZStd::string MaterialCanvasDocument::GetSymbolNameFromSlot(GraphModel::ConstSlotPtr slot) const
+    {
+        bool allowNameSubstitution = true;
+        if (auto dynamicNode = azrtti_cast<const AtomToolsFramework::DynamicNode*>(slot->GetParentNode().get()))
+        {
+            const auto& nodeConfig = dynamicNode->GetConfig();
+            AtomToolsFramework::VisitDynamicNodeSlotConfigs(
+                nodeConfig,
+                [&](const AtomToolsFramework::DynamicNodeSlotConfig& slotConfig)
+                {
+                    if (slot->GetName() == slotConfig.m_name)
+                    {
+                        allowNameSubstitution = slotConfig.m_allowNameSubstitution;
+                    }
+                });
+        }
+
+        if (!allowNameSubstitution)
+        {
+            return slot->GetName();
+        }
+
+        if (slot->SupportsExtendability())
+        {
+            return AZStd::string::format(
+                "%s_%s_%d", GetSymbolNameFromNode(slot->GetParentNode()).c_str(), slot->GetName().c_str(), slot->GetSlotSubId());
+        }
+
+        return AZStd::string::format("%s_%s", GetSymbolNameFromNode(slot->GetParentNode()).c_str(), slot->GetName().c_str());
     }
 
     AZStd::vector<AZStd::string> MaterialCanvasDocument::GetMaterialInputsFromSlot(
-        GraphModel::ConstNodePtr node, const AtomToolsFramework::DynamicNodeSlotConfig& slotConfig) const
+        GraphModel::ConstNodePtr node,
+        const AtomToolsFramework::DynamicNodeSlotConfig& slotConfig,
+        const AZStd::vector<AZStd::pair<AZStd::string, AZStd::string>>& substitutionSymbols) const
     {
         AZStd::vector<AZStd::string> materialInputsForSlot;
 
         if (auto slot = node->GetSlot(slotConfig.m_name))
         {
             AtomToolsFramework::CollectDynamicNodeSettings(slotConfig.m_settings, "materialInputs", materialInputsForSlot);
 
-            ReplaceStringsInContainer("SLOTSTANDARDSRGMEMBER", ConvertSlotValueToSrgMember(node, slotConfig), materialInputsForSlot);
-            ReplaceStringsInContainer("SLOTNAME", slot->GetName().c_str(), materialInputsForSlot);
-            ReplaceStringsInContainer("SLOTTYPE", ConvertSlotTypeToAZSL(slot->GetDataType()->GetDisplayName()), materialInputsForSlot);
-            ReplaceStringsInContainer("SLOTVALUE", ConvertSlotValueToAZSL(slot->GetValue()), materialInputsForSlot);
+            ReplaceSymbolsInContainer(substitutionSymbols, materialInputsForSlot);
+            ReplaceSymbolsInContainer("SLOTSTANDARDSRGMEMBER", GetAzslSrgMemberFromSlot(node, slotConfig), materialInputsForSlot);
```

Comment:
This line is missing from the earlier function that also does the name substitution, does it need to be added there?